CONTRIBUTING.md

```diff
@@ -66,6 +67,111 @@ please ensure that:
    deprecation, removal etc) that you have added a reno release note for that
    change and that the PR is tagged for the changelog.
 
+
+### Code review
+
+All code merged to Qiskit, even from maintainers, goes through a code-review
+process after a pull request is made.  We only have a small number of
+maintainers who can authorise a merge, and it's useful for everyone if the
+community can take part in helping make sure pull requests are technically
+correct, well tested, and easy to review.
```

Comment:
One thing I did in #6641 is to explicitly call out the value of people doing code review and also helping to reduce the maintainer burden. I think it would be good to reinforce this a bit more here. At least for me a big motivation in the original PR was to try and more strongly reinforce the value of having a culture of mutual review regardless of whether you empowered to merge code.

Reply:
How's ef1c3441a?

Reply:
Thanks, yeah that looks better to me.